internal/engine/build_and_deployer.go

```diff
@@ -22,3 +23,46 @@ type BuildAndDeployer interface {
 	// consider putting elsewhere or renaming (`Warm`?).
 	GetContainerForBuild(ctx context.Context, build BuildResult) (k8s.ContainerID, error)
 }
+
+type FirstLineBuildAndDeployer BuildAndDeployer
```

Comment:
it's a little bit weird that some places use the abstract name and some places use the concrete name. it feels like a half-finished abstraction

Reply:
@nicks is this more what you had in mind?